base_layer/core/src/transactions/transaction_components/encrypted_openings_s.rs

```diff
@@ -0,0 +1,211 @@
+// Copyright 2022 The Tari Project
+//
+// Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+// following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+// disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+// following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+// products derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+// USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE
+//
+// Portions of this file were originally copyrighted (c) 2018 The Grin Developers, issued under the Apache License,
+// Version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0.
+
+//! Encrypted openings using the the standard variant ChaCha20-Poly1305 encryption with a fixed zero nonce to save
+//! space - this design assumes a unique `encryption_key` every time and therefore uses a fixed nonce.
+
+use borsh::{BorshDeserialize, BorshSerialize};
+use chacha20poly1305::{
+    aead::{Aead, Error, Payload},
+    ChaCha20Poly1305,
+    KeyInit,
+    Nonce,
+};
+use digest::{generic_array::GenericArray, FixedOutput};
+use serde::{Deserialize, Serialize};
+use tari_common_types::types::{Commitment, PrivateKey};
+use tari_crypto::{hash::blake2::Blake256, hashing::DomainSeparatedHasher};
+use tari_utilities::{safe_array::SafeArray, ByteArray, ByteArrayError};
+use thiserror::Error;
+use zeroize::Zeroize;
+
+use super::EncryptedOpeningsKey;
+use crate::transactions::{tari_amount::MicroTari, TransactionKdfDomain};
+
+const VALUE_SIZE: usize = 8;
```

Comment:
You can define this as `std::mem::size_of::<u64>()` to avoid a magic number here.